pydantic/decorators.py

```diff
@@ -464,3 +464,10 @@ def dec(f: Callable[..., Any]) -> _decorators.PydanticDecoratorMarker[Any]:
         return dec
     else:
         return dec(__f)
+
+
+class AfterValidator:
+    def __init__(self, __func: _decorators.OnlyValueValidator | _core_schema.GeneralValidatorFunction) -> None:
+        self.func: _core_schema.GeneralValidatorFunction = _decorators.make_generic_v2_field_or_general_validator(
+            __func, mode='after'
+        )
```

Comment:
Is there a better place to put these?

Reply:
I guess we need a new module, since they're not dectorators.

We combine this with some of the other types designed to be used as annotations, like `Strict` and put them in `annoated_types.py` - we can add a docstring say these are pydantic specific extensions to the ideas of `annoated-types`.

Should we make this a dataclass?

I know we haven't used dataclasses elsewhere here, but I think we have in `types.py`, I don't see why we shouldn't.

Reply:
Sounds good. I can make this a dataclass. If we don't mind using dataclasses I think there's several other things that could be dataclasses, but I can start here

Reply:
absolutely. I guess we can have something like

```py
if whatever version:
     dc_kwargs = {'slots': Ture}
else:
    dc_kwargs = {}
```

So we can use slots on these dataclasses in a backwards compatible way, we might already do that in `types.py`

Reply:
Yup we do that in Annotated-types, I can just copy it